template.yaml

```diff
@@ -270,8 +272,9 @@ Resources:
                     - ec2:TerminateInstances
                   Resource: !Sub "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
                   Condition:
-                    StringLike:
-                      "ec2:ResourceTag/aws:autoscaling:groupName": !Sub "${AWS::StackName}-AgentAutoScaleGroup-*"
+                    StringEquals:
+                      "ec2:ResourceTag/Role": "buildkite-agent"
+                      "ec2:ResourceTag/aws:autoscaling:groupName": !Ref AgentAutoScaleGroup
```

Comment:
As with `ssm:GetCommandInvocation`, `ssm:DescribeInstanceInformation` does not support `Resource`s or `Condition`s. Unfortunately we have to grant `*` for that one.

The `Condition`s look fine for EC2 though 👍

Reply:
@lucaswilric `- ssm:DescribeInstanceInformation` is already pinned to `Resource: '*'` due to this issue, so I think we are fine here? 
Condition is applied to `ec2:TerminateInstances` Action.